pkg/v1/remote/transport/ping.go

```diff
@@ -108,8 +108,8 @@ func ping(ctx context.Context, reg name.Registry, t http.RoundTripper) (*pingRes
 			}, nil
 		case http.StatusUnauthorized:
 			if challenges := authchallenge.ResponseChallenges(resp); len(challenges) != 0 {
-				// If we hit more than one, I'm not even sure what to do.
-				wac := challenges[0]
+				// If we hit more than one, let's try to find one that we know how to handle.
+				wac := pickFromMultipleChallenges(challenges)
```

Comment:
What happens if we don't know how to handle any of the challenges?

I feel like maybe we should filter out unrecognized challenges and make sure `len(filtered) != 0`?

Reply:
If there's neither basic nor bearer challenge, the last challenge will be returned. This will lead to `unrecognized challenge` error [here](https://github.com/google/go-containerregistry/blob/main/pkg/v1/remote/transport/transport.go#L101).
I think I would leave checking if challenge can be handled outside ping.go. Here, we only have some preference as to which challenge to choose, but ultimately we can return one that's not basic nor bearer.

Reply:
I don't think it will return the last challenge because the `wac` in `for _, wac := range challenges` will shadow the outer `wac` variable. Instead, it would return a zero-value Challenge.

Maybe if we don't encounter a useful challenge, we just return the first one? (line 148 currently)

Reply:
Ah, right, my bad, it should be `for _, wac = range challenges` so `wac` is not shadowed... But ok, I will change it so it returns the first challenge if neither basic nor bearer are found.

Reply:
Thanks for noticing that!